native/cocos/platform/linux/modules/SystemWindow.cpp

```diff
@@ -38,67 +38,67 @@
 #include "platform/SDLHelper.h"
 
 namespace cc {
-SystemWindow::SystemWindow(IEventDispatch *delegate)
-: _sdl(std::make_unique<SDLHelper>(delegate)) {
+SystemWindow::SystemWindow(uint32_t windowId, void *externalHandle)
+: _windowId(windowId) {
+    if (externalHandle) {
+        _windowHandle = reinterpret_cast<uintptr_t>(externalHandle);
+    }
 }
 
 SystemWindow::~SystemWindow() {
-}
-
-int SystemWindow::init() {
-    return _sdl->init();
-}
-
-void SystemWindow::pollEvent(bool *quit) {
-    return _sdl->pollEvent(quit);
+    _windowHandle = 0;
+    _windowId = 0;
 }
 
 void SystemWindow::swapWindow() {
-    _sdl->swapWindow();
+    SDLHelper::swapWindow(_window);
 }
 
 bool SystemWindow::createWindow(const char *title,
                                 int w, int h, int flags) {
-    _sdl->createWindow(title, w, h, flags);
-    _width = w;
-    _height = h;
+    if (!createWindow(title, -1, -1, w, h, flags)) {
+        return false;
+    }
     return true;
 }
 
 bool SystemWindow::createWindow(const char *title,
                                 int x, int y, int w,
                                 int h, int flags) {
-    // Create window
-    _sdl->createWindow(title, x, y, w, h, flags);
-    _width = w;
+    if (x == -1 && y == -1) {
+        _window = SDLHelper::createWindow(title, w, h, flags);
+    } else {
+        _window = SDLHelper::createWindow(title, x, y, w, h, flags);
```

Comment:
Can work if `x == -1` and `y != -1`?

Reply:
What happen if `x == -10`?

Reply:
-1 means use a system decided postion. @qiuguohua

Reply:
@PatriceJiang

Reply:
> BaseGame.cpp is supposed to be for the editor?

It is the base class of game logic.